manager/integration/tests/test_migration.py

```diff
@@ -225,22 +232,135 @@ def test_migration_with_rebuilding_replica(clients, volume_name):  # NOQA
 
 @pytest.mark.coretest  # NOQA
 @pytest.mark.migration # NOQA
-@pytest.mark.skip(reason="TODO") # NOQA
-def test_migration_with_restore_volume(clients, volume_name):  # NOQA
+def test_migration_with_restore_volume(core_api, client, clients, volume_name, storage_class, pvc_name, set_random_backupstore):  # NOQA
     """
     Test that a restored volume can be migrated.
 
     1. Prepare one backup.
     2. Create a StorageClass with `migratable` being enabled and
        `fromBackup` pointing to the above backup.
     3. Create a new RWX migratable volume using the StorageClass.
-    2. Attach to node 1, then write some data.
-    4. Attach the volume to node 2 (migration target).
-    5. Wait for the migration ready. Verify that field
+    4. Attach to node 1, then write some data.
+    5. Attach the volume to node 2 (migration target).
+    6. Wait for the migration ready. Verify that field
        `volume.controllers[0].requestedBackupRestore` is empty.
-    6. Confirm the migration then validate the data.
+    7. Confirm the migration then validate the data.
     """
-    return
+    # Step 1
+    lht_host_id = get_self_host_id()
+    volume = create_and_check_volume(client,
+                                     volume_name,
+                                     REPLICA_COUNT,
+                                     size=str(DEFAULT_VOLUME_SIZE * Gi))
+
+    volume.attach(hostId=lht_host_id)
+    volume = common.wait_for_volume_healthy(client, volume_name)
+
+    bv, b, _, _ = create_backup(client, volume_name)
+
+    # Step 2
+    storage_class['metadata']['name'] = "longhorn-from-backup"
+    storage_class['parameters']['fromBackup'] = b.url
+    storage_class['parameters']['migratable'] = "true"
+
+    create_storage_class(storage_class)
+
+    # Stpe 3
+    VOLUME_SIZE = str(DEFAULT_VOLUME_SIZE * Gi)
+
+    backup_pvc_name = generate_volume_name()
+
+    backup_pvc_spec = {
+        "apiVersion": "v1",
+        "kind": "PersistentVolumeClaim",
+        "metadata": {
+                "name": backup_pvc_name,
+        },
+        "spec": {
+            "accessModes": [
+                "ReadWriteMany"
+            ],
+            "storageClassName": storage_class['metadata']['name'],
+            "resources": {
+                "requests": {
+                    "storage": VOLUME_SIZE
+                }
+            }
+        }
+    }
+
+    volume_count = len(client.list_volume())
+
+    core_api.create_namespaced_persistent_volume_claim(
+        'default',
+        backup_pvc_spec
+    )
+
+    backup_volume_created = False
+
+    for i in range(RETRY_COUNTS):
+        if len(client.list_volume()) == volume_count + 1:
+            backup_volume_created = True
+            break
+        time.sleep(RETRY_INTERVAL)
```

Comment:
Why not just reuse `wait_for_volume_creation`?

Remember reusing shared functions if possible, otherwise there will be a pain later because of duplicated codes anywhere.

Reply:
Hi @innobead , 
I added `wait_for_volume_creation` in new created common function `create_rwx_migratable_volume_using_storageslass`